daemon/disk_usage.go

```diff
@@ -34,6 +29,10 @@ func (daemon *Daemon) ContainerDiskUsage(ctx context.Context) ([]*types.Containe
 				Status:     c.Status,
 				Mounts:     c.Mounts,
 			})
+			lCtx.idx++
+			return true
+		}); err != nil {
+			return nil, fmt.Errorf("failed to range over containers: %v", err)
```

Comment:
I need to take some time looking at this commit to get the full picture.

Just a quick comment on this one; wondering if this is more descriptive than the old error.

One thing I _did_ notice is that we're masking the underlying error type if the filter is invalid; `reduceContainers` / `rangeContainers` calls `config.Filters.Validate`, which can return an `errdefs.InvalidParameter`. Using `fmt.Errorf` here will change a `4xx` status to a `5xx` status if that happens.